opac/webapp/controllers.py

```diff
@@ -853,6 +856,104 @@ def get_article_by_aid(aid, journal_url_seg, lang=None, gs_abstract=False, **kwa
     return article
 
 
+def _articles_or_abstracts_sorted_by_order_or_date(iid, gs_abstract=False):
+    """
+    Retorna uma lista de artigos de um _fascículo_ ou de um _bundle_
+
+    - ``iid``: chave primaria de número para escolher os artigos.
+    - ``kwargs``: parâmetros de filtragem.
+
+    Em caso de não existir itens retorna {}.
+
+    """
+    articles = get_articles_by_iid(iid, is_public=True)
+    if gs_abstract:
+        # FIXME - Melhorar esta consulta
+        # conseguir em um único comando
+        # obter só os documentos que contenham resumo
+        return [a for a in list(articles) if a.abstracts]
```

Comment:
Se passar o argumento `abstracts__exists=True` em `get_articles_by_iid()`, a consulta retorna somente os artigos que possuem o campo.

Reply:
@patymori obrigada, vou testar, perdi bastante tempo tentando descobrir como fazer e não encontrei

Reply:
@patymori retorna incorretamente porque o "abstracts" existe mas é uma lista vazia, então não deveria retornar.

Reply:
@patymori eu acho que consegui. Vou fazer mais testes.

Reply:
@robertatakenaka, vi que teria problemas mesmo. Uma outra alternativa seria fazer uma [raw query](http://docs.mongoengine.org/guide/querying.html#raw-queries) mas aguardo teus testes.

Reply:
@robertatakenaka fiz um outro teste aqui, alterando o documento no MongoDB, e quando o `abstracts` for uma lista vazia, a consulta que passei inicialmente não trás o documento.